docs/website/01_framework/assembler.md

```diff
@@ -0,0 +1,89 @@
+- `fes` is a container that contains all the finite elements that should be assembled.
+- `dirichletFlags` is a `#!cpp std::vector<bool>` type. The `i`-th degree of freedom is fixed when `#!cpp dirichletFlags[i] = true`.
+  When a reduced matrix or vector is chosen, the corresponding row and column entries are removed. 
+
+## FlatAssemblerBase
+The FlatAssemblerBase is the base for all assemblers currently available. All other assemblers inherit from this one, 
+i.e., their interface includes the following functions:
+```cpp
+size_t size() // (1)
+size_t reducedSize() // (2)
+auto &finiteElements() const // (3)
+Eigen::VectorXd createFullVector(const Eigen::VectorXd &reducedVector) // (4)
+size_t constraintsBelow(size_t i) // (5)
+bool isConstrained(size_t i) // (6)
+size_t estimateOfConnectivity() // (7)
+```
+
+1. Returns the number of degrees of freedom.
+2. Returns the number of degrees of freedom that are not constrained by a Dirichlet boundary condition.
+3. Returns a reference to the finite element container, which was passed to the assembler.
+4. Gets a reduced vector and returns a full vector. Entries corresponding to fixed dofs are set to 0. The values of the other entries are
+    obtained from the reduced vector.
+5. Indicates how many degrees of freedom {0,1,...i-1} are fixed.
+6. Indicates whether the degree of freedom `i` is fixed.
+7. Returns 8x the number of grid elements, which is an estimate for the connectivity. It can be used to allocate vectors.
+
+
+## ScalarAssembler
+It has the capabilities of [FlatAssemblerBase](#flatassemblerbase) plus one additional function:
+```cpp
+double& getScalar(const RequirementType& fErequirements)
+```
+This assembler can be used when only a scalar quantity is of interest and the assembly of matrices or vectors is irrelevant.
+The available requirements are explained on the [FE requirements page](feRequirements.md).
+`dirichletFlags` is not used in this assembler.
+
+It assembles the requested scalar quantity. A call to this function could look like this:
+```cpp
+ScalarAssembler myAssembler(...) // (1)
+const auto& K = myAssembler.getScalar(energy) // (2)
+```
+
+1. Represents the construction of the desired assembler.
+2. To learn more about the available alternatives to `energy` and how this works, read the [FE requirements](feRequirements.md) page.
+
+## VectorFlatAssembler
+It has all the features of [ScalarAssembler](#scalarassembler) plus more, like:
+```cpp
+Eigen::VectorXd& getVector(const RequirementType& fErequirements)
+Eigen::VectorXd& getReducedVector(const RequirementType& fErequirements)
+```
+As the name suggests, the full vector or a reduced vector considering boundary conditions is returned.
```

Comment:
The getVector writes zeros on the constraint terms

Reply:
This should be said here i think.

Reply:
Anyway the assemblers hould have a getRawVector functionality. I will look into adding this.

Reply:
There is an issue already #110